src/dataLoaders/components/collectorsWizard/TelegrafUIRefreshWizard.tsx

```diff
@@ -0,0 +1,178 @@
+// Libraries
+import React, {PureComponent} from 'react'
+import Loadable from 'react-loadable'
+import {connect, ConnectedProps} from 'react-redux'
+import {withRouter, RouteComponentProps} from 'react-router-dom'
+
+// Components
+import {ErrorHandling} from 'src/shared/decorators/errors'
+import {Overlay} from '@influxdata/clockface'
+import {PluginCreateConfigurationFooter} from 'src/writeData/components/PluginCreateConfigurationFooter'
+
+const spinner = <div />
+const TelegrafUIRefreshStepSwitcher = Loadable({
+  loader: () =>
+    import(
+      'src/dataLoaders/components/collectorsWizard/TelegrafUIRefreshStepSwitcher'
+    ),
+  loading() {
+    return spinner
+  },
+})
+
+// Actions
+import {notify as notifyAction} from 'src/shared/actions/notifications'
+import {
+  clearSteps,
+  decrementCurrentStepIndex,
+  incrementCurrentStepIndex,
+  setBucketInfo,
+  setCurrentStepIndex,
+  setSubstepIndex,
+} from 'src/dataLoaders/actions/steps'
+
+import {clearDataLoaders} from 'src/dataLoaders/actions/dataLoaders'
+
+// Types
+import {AppState, Bucket, ResourceType} from 'src/types'
+import {PluginCreateConfigurationStepProps} from 'src/writeData/components/PluginCreateConfigurationWizard'
+
+// Selectors
+import {getAll} from 'src/resources/selectors'
+import {getOrg} from 'src/organizations/selectors'
+
+// Utils
+import {isSystemBucket} from 'src/buckets/constants'
+
+type ReduxProps = ConnectedProps<typeof connector>
+type Props = ReduxProps & RouteComponentProps<{orgID: string}>
+
+@ErrorHandling
+class TelegrafUIRefreshWizard extends PureComponent<Props> {
+  public state = {pluginConfig: '', isValidConfiguration: false}
+
+  public componentDidMount() {
+    const {bucket, buckets} = this.props
+    if (!bucket && buckets && buckets.length) {
```

Comment:
```suggestion
    if (!bucket && Array.isArray(buckets) && buckets.length) {
```

_probably_ better to use an array check, rather than a falsy check, just for self documenting code purposes.

Reply:
👍